src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -867,18 +702,24 @@ def test_automation_job_create_commit_failed_check(
     treestatusdouble,
     get_automation_worker,
     monkeypatch,
-    failed_check_action_reason: Callable,
-    bad_action_reason: str,
+    failed_check_reason: Callable,
+    bad_action_type: str,
+    check_diff: Callable,
+    extract_email: Callable,
 ):
-    bad_action, reason = failed_check_action_reason(bad_action_reason)
+    bad_action, reason = failed_check_reason(bad_action_type)
+    # Warning: I suspect, but haven't been able to confirm, that doing this
+    # may directly modify the data in the fixture. If, in the future, the
+    # failed_check_reason fixtures appears to contain weird unexpected content in other
+    # test cases, look with suspicion towards this.
+    bad_action["action"] = "create-commit"
+    bad_action["date"] = 0
+    bad_action["diff"] = check_diff(bad_action_type)
```

Comment:
Would be better to pass these as arguments to the fixture, and override them in the fixture instead (common idiom). `bad_action_type` is passed twice, once to `failed_check_reason` which returns `bad_action`, and then again to override the value in `bad_action["diff"]`. This can be cleaned up I think.

Reply:
I got myself into a position where I have slightly different needs from the fixture, so the second need extends the first one... But I think I can just create a second fixture to hide this logic in.